styx-scheduler-service/src/main/java/com/spotify/styx/flyte/FlytePhase.java

```diff
@@ -0,0 +1,57 @@
+/*-
+ * -\-\-
+ * Spotify Styx Scheduler Service
+ * --
+ * Copyright (C) 2016 - 2020 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.styx.flyte;
+
+import com.google.common.collect.ImmutableMap;
+import flyteidl.core.Execution;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+public enum FlytePhase {
+  QUEUED,
+  RUNNING,
+  SUCCEEDING,
+  SUCCEEDED,
+  FAILING,
+  FAILED,
+  ABORTED,
+  TIMED_OUT;
+
+  private final static Map<Execution.WorkflowExecution.Phase, FlytePhase> phaseMapper =
+    ImmutableMap.<Execution.WorkflowExecution.Phase, FlytePhase>builder()
```

Comment:
Just as a side note. Since Java 9 `Map.of()` achieve the same result as `ImmutableMap`. No need to change it as we depend on Guava anyways

Reply:
👍